sdl-snowflake/src/main/scala/io/smartdatalake/workflow/connection/SnowflakeConnection.scala

```diff
@@ -69,14 +71,31 @@ case class SnowflakeTableConnection(override val id: ConnectionId,
           "sfWarehouse" -> warehouse
         )
       case _ => throw new IllegalArgumentException(s"($id) No supported authMode given for Snowflake connection.")
-    } else throw new IllegalArgumentException(s"($id) No authMode given for Snowflake connection.")
+    }
   }
 
-  override def factory: FromConfigFactory[Connection] = SnowflakeTableConnection
+  def getSnowparkSession(schema: String): Session = {
```

Comment:
It seems to me that the snowpark session is not yet shared, as every call to getSnowparkSession creates a new session. Also it seems that a snowpark session is schema dependent. You probably need to maintain a map of created sessions per schema and return already created sessions if existing...

Reply:
Every call to does create a new session, but the session is cached in SnowflakeTableDataObject._snowparkSession.

You need to pass a schema to create a Snowpark session, but you are not limited to that schema when using the session. It merely sets a default schema for the session. I always use fully qualified table names when interacting with Snowflake (via the Spark adapter or with Snowpark), so the default session is never used.